core/common/src/internal/-RealBufferedSource.kt

```diff
@@ -39,6 +41,7 @@ internal inline fun RealSource.commonRead(sink: Buffer, byteCount: Long): Long {
   return buffer.read(sink, toRead)
 }
 
+@OptIn(DelicateIoApi::class)
 internal inline fun RealSource.commonExhausted(): Boolean {
   check(!closed) { "closed" }
```

Comment:
Exception message could be a bit more verbose probably

Reply:
I'll update it to `Source is closed.` / `Sink is closed.`. Not sure if there is anything else we can say to a user about the problem.